### DISCUSSION TITLE
Migration to single fetch crashes for end users

### DISCUSSION THREAD

**Discussion by jansedlon** [remix-run/remix#10106] [Q&A]
Hello,

I have tried to enable single fetch on version 2.13.1 and migrate a single route to single fetch. That means i removed `defer` function from loader.

Immediately after deployment i was getting Sentry errors. The first one seems to make sense because frontend keeps the cached version and for some time is sending data requests on the backend

<img width="825" alt="image" src="https://github.com/user-attachments/assets/02cec974-c9ef-45ad-af84-a2065c78bb36">

However in the same trace it showed that it crashed because it couldn't load loader data.

<img width="842" alt="image" src="https://github.com/user-attachments/assets/b1834a90-dc0e-49d7-b7c0-02b1d3311bd4">

Is this because i deployed the single fetch and loader migration in a single deployment? Is there a way to not crash it for the end users? 😅

**Comment by jansedlon** [comment]
Anyone?

**Comment by jarvis394** [comment]
Your `useLoaderData` probably returns an object which fields are promises that resolve to the actual data. I don't think you are able to access those fields before the promise resolves, hence the `undefined` in Sentry logs.

Does your TS typecheck bring any errors in `app/routes/user+/stores+/details/route.tsx`?

Refer to docs: https://remix.run/docs/en/main/guides/streaming#2-component-setup

**Reply by jansedlon** [reply]
It doesn't, because `stores` property is not a Promise. I think there's an issue with updating app to single fetch and migrating routes in a single deployment. Frontend is typically cached for end users because Remix fingerprints assets. When new deployment is release, backend returns data in a new format but frontend doesn't yet know that single fetch is enabled.

However i'm not sure if there's a way around that other than enabling single fetch in a separate deployment from migrating existing routes.
